project_redss/translate_rapid_pro_keys.py

```diff
@@ -49,13 +53,46 @@ class TranslateRapidProKeys(object):
     WEEK_4_START = isoparse("2018-12-23T00:00:00+03:00")
 
     @classmethod
-    def translate_rapid_pro_keys(cls, user, data):
+    def build_message_to_s01e02_dict(cls, user, data, coda_input_dir):
+        # Duplicate the input list because reading the file requires appending data to the TracedData,
+        # and we don't actually want to modify the input at this stage of the pipeline.
+        data = [td.copy() for td in data]
+
+        # Apply the week 3 codes from Coda.
+        message_id_key = "radio_show_3_message_id"
+        coded_ws_key = "radio_show_3_ws"
+        TracedDataCoda2IO.add_message_ids(user, data, cls.WEEK_3_VALUE_KEY, message_id_key)
+        coda_input_path = path.join(coda_input_dir, "s01e03.json")
+        with open(coda_input_path) as f:
+            TracedDataCoda2IO.import_coda_2_to_traced_data_iterable(
+                user, data, message_id_key, {coded_ws_key: CodeSchemes.WS_CORRECT_DATASET}, f)
+
+        # Parse the loaded codes into a look-up table of raw message string -> is ws boolean.
+        message_to_ws_dict = dict()
         for td in data:
-            mapped_dict = dict()
+            label = td.get(coded_ws_key)
+            if label is not None:
+                message_to_ws_dict[td.get(cls.WEEK_3_VALUE_KEY)] = \
+                    label["CodeID"] == CodeSchemes.WS_CORRECT_DATASET.get_code_with_match_value("s01e02").code_id
+
+        return message_to_ws_dict
 
-            # Redirect any week 4 messages which were in the week 3 flow due to a late flow change-over.
+    @classmethod
+    def translate_rapid_pro_keys(cls, user, data, coda_input_dir):
+        # Build a map of raw week 3 messages to wrong scheme data
+        message_to_s01e02_dict = cls.build_message_to_s01e02_dict(user, data, coda_input_dir)
```

Comment:
Please could you put a note to explicitly denote a work around to an incident

One of the things that is unclear in reading a project like this for a new SWE is what is 'expected everywhere' and what is 'exigent circumstances'